src/nvim/if_cscope.c

```diff
@@ -1772,11 +1777,14 @@ static int cs_read_prompt(int i)
       if (bufpos < maxlen - 1 && vim_isprintc(ch)) {
         // lazy buffer allocation
         if (buf == NULL) {
-          buf = xmalloc(maxlen);
+          assert(maxlen >= 0);
+          buf = xmalloc((size_t)maxlen);
         }
         {
           /* append character to the message */
-          buf[bufpos++] = ch;
+          assert(ch >= 0);
+          assert(ch <= CHAR_MAX);
```

Comment:
> And the core dump will have the stack values, so the granularity isn't needed, I think.

While much simpler (in some places) than Vim, it does seem like Nvim is too complicated to get much meaning out of an assert. Still, I don't see any reason to use `&&`, besides the line savings (which IMO doesn't matter because they're harder to read).